src/citrine/resources/predictor.py

```diff
@@ -96,38 +128,69 @@ def update(self, predictor: Predictor) -> Predictor:
     def _train(self, uid: Union[UUID, str]):
         path = self._predictors_path("train", uid)
         entity = self.session.put_resource(path, {}, version=self._api_version)
-        return self.build(entity)
+        predictor = self.build(entity)
+        return self._inject_archive_info(predictor)
 
-    def archive(self, uid: Union[UUID, str]):
-        """Archive a predictor.
+    def archive_root(self, uid: Union[UUID, str]):
+        """Archive a root predictor.
 
         uid: Union[UUID, str]
             Unique identifier of the predictor to archive.
 
         """
         path = self._predictors_path("archive", uid)
-        entity = self.session.put_resource(path, {}, version=self._api_version)
-        return self.build(entity)
+        self.session.put_resource(path, {}, version=self._api_version)
```

Comment:
I don't necessarily agree with always returning an object, but you are changing the pattern here.

Reply:
That's part of why it's a new method.